manager/eris-mint/eris-mint.go

```diff
@@ -185,15 +185,14 @@ func (app *ErisMint) Commit() (res tmsp.Result) {
 	// sync the AppendTx cache
 	app.cache.Sync()
 
-	// if there were any txs in the block,
-	// reset the check cache to the new height
-	if app.nTxs > 0 {
-		log.WithFields(log.Fields{
-			"txs": app.nTxs,
-		}).Info("Reset checkCache")
-		app.checkCache = sm.NewBlockCache(app.state)
-	}
-	app.nTxs = 0
+	// Refresh checkCache even if there were no transactions. A CheckTx may have
```

Comment:
The concurrency model of TMSP application interface is still troublesome and we need to enforce it from the application side.  However, is the mempool not supposedly locked during commit, such that CheckTx and AppendTx are not called?

Can we consider a lock controlled by commit on holding all CheckTx/AppendTx, and holding commit until all CheckTx and AppendTx have cleared out of the queue?  -- I ve held off on this because we need to revisit the async model here at a fundamental level.

I am concerned here that undoing the nTxs breaks the original assumptions in this code.